### DISCUSSION TITLE
BatchGetItem Java example with single partition key with different partition key value

### DISCUSSION THREAD

**Discussion by banandh** [aws/aws-sdk-java#3083] [Q&A]
I was using below option 
BatchGetItemSpec batchGetItemSpec = new BatchGetItemSpec();
            BatchGetItemRequest batchGetItemRequest = new BatchGetItemRequest();
            QuerySpec querySpec = new QuerySpec();

            Map<String, KeysAndAttributes> requestItem = new HashMap<>();

            Map<String, AttributeValue> attributeValueMap = new HashMap<>();
             for (GetScoopService.GeneratedKeys key: keys) {
                 attributeValueMap.put("pk",new AttributeValue(getPartitionKey(key)));

             }
            requestItem.put(sskTableName,new KeysAndAttributes().withKeys(attributeValueMap));

            batchGetItemRequest.setRequestItems(requestItem);

But here " attributeValueMap.put("pk",new AttributeValue(getPartitionKey(key)));" I m adding this in map so its overriding everything one single item is requested. I need to reqeust a single table against a PK for different PK values in batch query. PLease share some java samples

**Comment by debora-ito** [comment]
So keys in `KeysAndAttributes` takes a Collection: `new KeysAndAttributes().withKeys(<Collection>)`

It'd look something like this (based on your sample code):

```java
        for (GetScoopService.GeneratedKeys key: keys) {
            listOfKeys.add(Map.of("pk", new AttributeValue(getPartitionKey(key))));
        }

        requestItem.put(tableName, new KeysAndAttributes().withKeys(listOfKeys));
        batchGetItemRequest.setRequestItems(requestItem);
```
